virttest/qemu_devices/qdevices.py

```diff
@@ -927,8 +927,22 @@ def verify_hotplug(self, out, monitor):
         return False
 
 
-class Dimm(QDevice):
+class Nvdimm(QDevice):
+    """
+    NVDIMM device, using the QOM objects 'memory-backend=file', not
+    support hotplug yet.
+    """
+    __attributes__ = ["memdev", "slot", "addr", "node"]
+
+    def __init__(self, params=None):
```

Comment:
view from your code, nvdimm should another kind of dimm device, so I think better ways is redfine __init__() for DIMM devices.